### PR TITLE
[Feature request]make some providers optional

### PR DESCRIPTION
**Pull request by snylonue**

### CODE REVIEW

File: anni-provider/Cargo.toml [hunk 1]
```diff
@@ -9,17 +9,17 @@ repository = "https://github.com/ProjectAnni/anni"
 license = "Apache-2.0"
 
 [dependencies]
-tokio = { version = "1", features = ["time"] }
+tokio = { version = "1", features = ["time", "fs", "rt"] }
```
Comment (Yesterday17) [inline anni-provider/Cargo.toml:12]: why fs and rt are added?

Comment (snylonue) [reply anni-provider/Cargo.toml:12]: otherwise the compilation without some features will fail due to the use of `tokio::fs` and  `tokio::spawn`

Comment (Yesterday17) [reply anni-provider/Cargo.toml:12]: Add `tokio/xxxx` to corresponding features? But why the compilation did not fail before(?

File: anni-provider/Cargo.toml [hunk 2]
```diff
@@ -33,3 +33,12 @@ anni-flac = { path = "../anni-flac", version = "0.2.2", features = ["async"] }
 version = "0.11"
 default-features = false
 features = ["json", "rustls-tls", "stream"]
+optional = true
+
+[features]
+default = ["convention", "drive", "proxy", "strict"]
```
Comment (Yesterday17) [inline anni-provider/Cargo.toml:39]: add `full` feature and use it as default?

File: anni-provider/Cargo.toml (hunk 1 above)
Comment (snylonue) [reply anni-provider/Cargo.toml:12]: > But why the compilation did not fail before(?

IMO other dependencies enabled these features but they are now optional.

File: anni-provider/Cargo.toml (hunk 2 above)
Comment (snylonue) [reply anni-provider/Cargo.toml]: added

File: anni-provider/Cargo.toml (hunk 1 above)
Comment (Yesterday17) [reply anni-provider/Cargo.toml:12]: I think fs is reasonable, but rt seems that we're asking a library to depend on a runtime, which is confusing to me.

(Might not be correct. I'm not familiar with this part.)

Comment (Yesterday17) [reply anni-provider/Cargo.toml:12]: Digged deeper, and the `rt` feature looks reasonable to me now.

**Pull request merged**